Remove depricated start_decompisition function in the python class of decomp_non_unitary

jnadori opened the pull request.

rakytap commented:
Nice catch. The file examples/decomposition/example was initially designed for the adaptive algorithm with CRY gates. Please restore the original file with the adaptive interface, instead of the non-unitary algo. An example with the non-unitary interface should be in a new file, for example: example_tree_search_decomposition.py. (The non-unitary interface will be renamed to tree search at the end.)

jnadori commented:
Noted. Original file restored and fixed example script is now moved to a different file. I have also found another bug in gates block get parameter max similar to the one found in list gates where the parameter order hasn't been reverted back to the current order. Important to not that the DFE portion of gates block still operates with the reversed parameter order, is this by design?
